### PR TITLE
Added support for baked collision checking.

### PR DESCRIPTION
**Pull request by mkoval**
This depends on https://github.com/personalrobotics/prpy/pull/343 and https://github.com/personalrobotics/prpy/pull/345.

### CODE REVIEW

File: src/or_trajopt/planning.py [hunk 1]
```diff
@@ -58,7 +59,8 @@ class CostType(enum.Enum):
 
 
 class TrajoptWrapper(MetaPlanner):
-    def __init__(self, planner):
+    def __init__(self, planner,
+            robot_collision_checker=DefaultRobotCollisionChecker):
```
Comment (psigen) [inline src/or_trajopt/planning.py]: Fix indentation here

File: src/or_trajopt/planning.py [hunk 2]
```diff
@@ -164,7 +168,7 @@ def _addFunction(problem, timestep, i_dofs, n_dofs, fnargs):
             ValueError('Invalid cost or constraint type: {:s}'
                        .format(str(fntype)))
 
-    def _Plan(self, robot, request,
+    def _Plan(self, robot, robot_checker, request,
```
Comment (psigen) [inline src/or_trajopt/planning.py:178]: Why do the internal methods all pass `robot_checker`, but the public methods all use the `self` reference?

Comment (mkoval) [inline src/or_trajopt/planning.py:178]: `self.robot_collision_checker` is a factory that creates a `robot_checker` context manager. This additional level of indirection allows us to do baking in `__enter__` and cleanup baking in `__exit__`. It also lets us wrap the `__enter__` and `__exit__` methods of a `CollisionOptionsStateSaver` to enable `CollisionOptions.ActiveDOFs`.

All of the internal methods take `robot_checker` so we only do the baking _once_ per query. The usage looks pattern looks like this:

``` python
with self.robot_collision_checker(robot) as robot_checker:
    return self._Plan(robot, robot_checker, request, **kwargs)
    # _Plan internally forwards robot_checker to other functions.
```

The name `robot_collision_checker` and `robot_checker` is confusing, but is chosen to be consistent with the other planners in PrPy. @cdellin and I discussed this in https://github.com/personalrobotics/prpy/pull/343, but did not come up with a better name. I would prefer to stick with these names for now and refactor everything (including all of the planners in PrPy) at once.

File: src/or_trajopt/planning.py (hunk 1 above)
Comment (mkoval) [inline src/or_trajopt/planning.py]: What do you want the indentation to look like? I am fine with changing it, but it looks correct to me. 😕

**Pull request merged**